### PR TITLE
Main & postee API merge

### PR DESCRIPTION
**Pull request by elad-da**

### CODE REVIEW

**Comment by simar7** [comment]
Hi - what issue is this addressing?

**Comment by AndreyLevchenko** [comment]
it allows to use postee as library to send notifications outside:
See example of usage: https://github.com/elad-da/postee/blob/main-api-merge/router/api_integration_test.go
not sure why it's going to dev branch.

**Comment by elad-da** [comment]
We created a dev branch so we can align with the main branch and keep changing the Postee to match our requirements to work as a library, without risking the main branch.

File: outputs/email.go [hunk 1]
```diff
@@ -101,12 +117,12 @@ func sendViaMxServers(from, port, subj, msg string, recipients []string) {
 					"Subject: %s\r\n"+
 					"Content-Type: text/html; charset=UTF-8\r\n\r\n%s\r\n", rcpt, from, subj, msg)
 
-			if err := smtp.SendMail(mx.Host+":"+port, nil, from, []string{rcpt}, []byte(message)); err != nil {
-				log.Printf("SendMail error to %q via %q", rcpt, mx.Host)
-				log.Print(err)
+			if err := smtp.SendMail(mx.Host+":25", nil, from, []string{rcpt}, []byte(message)); err != nil {
```
Comment (AndreyLevchenko) [inline outputs/email.go:120]: it looks like reverting some fix we just made

**Review by AndreyLevchenko — CHANGES_REQUESTED** [review]

**Review by simar7 — CHANGES_REQUESTED** [review]
> We created a dev branch so we can align with the main branch and keep changing the Postee to match our requirements to work as a library, without risking the main branch.

hi @tomyweiss and @elad-da - appreciate your contributions but as a new reviewer & maintainer of Postee I'm unable to understand the scope of this PR. It's very difficult to review such a PR with many conflicts as @AndreyLevchenko mentioned. 

Could you rebase on the latest main branch of Postee and open a new PR? Happy to assist you offline or here if need be. Let us know!

**Comment by elad-da** [comment]
Hi @simar7, this is a rebase on the main branch of the Postee-Api feature (postee as a library).

**Review by AndreyLevchenko — APPROVED** [review]

**Pull request merged**